### PR TITLE
Add RGI_Emoji output

### PR DESCRIPTION
**Pull request by mathiasbynens**
In the same vein as #71, this patch reduces the number of hardcoded Unicode properties in the source code by leveraging the new `RGI_Emoji` property as much as possible.

This brought to light some more inconsistencies between the Unicode Standard's definitions and real-world implementations, similar to those that have previously been discussed in this repo's issue tracker.

For example, U+1F575 U+FE0F is a qualified emoji per `emoji-sequences.txt`:

```
1F575 FE0F    ; Basic_Emoji                  ; detective                                                      # E0.7   [1] (🕵️)
```

But without the U+FE0F, it's not, even though it still renders as an emoji on macOS — and so folks might expect emoji-regex to match it. Our previous regex matched it, which is wrong per spec but debatably "right" per macOS.

### CODE REVIEW

**Comment by mathiasbynens** [comment]
For backwards compatibility, I've added a new output target for the new `RGI_Emoji` regular expression (instead of updating the “main” one).

**Review by scriby — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -49,15 +49,35 @@ Matched sequence 👩🏿 — code points: 2
...
+// emoji recommended for general interchange, as defined via the
+// `RGI_Emoji` property in the Unicode Standard.
+// https://unicode.org/reports/tr51/#def_rgi_set
+// When in doubt, use this!
+const emojiRegexRGI = require('emoji-regex/RGI_Emoji.js');
+
+// This is the old regular expression, prior to `RGI_Emoji` being
+// standardized. In addition to all `RGI_Emoji` sequences, it matches
+// some emoji you probably don’t want to match (such as emoji component
+// symbols that are not meant to be used separately).
```
Comment (rsheeter) [inline README.md:67]: When *should* you use this one? - comment reads as if answer is never

File: README.md [hunk 2]
```diff
@@ -49,15 +49,35 @@ Matched sequence 👩🏿 — code points: 2
...
+const emojiRegex = require('emoji-regex/index.js');
+
+// This regular expression matches even more emoji than the previous
+// one, including emoji that render as text instead of icons (i.e.
+// emoji that are not `Emoji_Presentation` symbols and that aren’t
+// forced to render as emoji by a variation selector).
+const emojiRegexText = require('emoji-regex/text.js');
```
Comment (rsheeter) [inline README.md:74]: Similar to the one above, when should this be used?

Comment (mathiasbynens) [reply README.md:74]: It should only be used by those seeking 100% backwards-compatibility with older emoji-regex releases. I plan on removing it in the next major release.

File: README.md (hunk 1 above)
Comment (mathiasbynens) [reply README.md:67]: It should only be used by those seeking 100% backwards-compatibility with older emoji-regex releases. I plan on removing it in the next major release.

File: README.md (hunk 2 above)
Comment (mathiasbynens) [reply README.md:74]: The `text` variant was added after popular demand: folks wanted a regex that matched more "emoji" than the other regex (even when technically they aren't pictographic emoji, nor guaranteed to be rendered as emoji per spec).

File: test/tests.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
 const assert = require('assert');
```
[the remark this answers is not in the document]
Comment (mathiasbynens) [reply test/tests.js:1]: I'll add some tests for things like lone regional indicators, or regional indicators in non-RGI combinations, etc.

**Pull request merged**